starter-code/src/ContactManager.jsx

```diff
@@ -3,42 +3,54 @@ import React from 'react'
 class ContactManager extends React.Component {
 
   state = {
-    contacts : this.props.contacts
+    contactsPool : this.props.contacts,
+    contacts : this.props.contacts.splice(0, 5)
   }
 
-  // onInputChange = (e) => {
-  //   this.setState({newUser : e.target.value});
-  // }
-  //
-  // onButtonClick = (e) => {
-  //   if(!this.state.newUser) { return; }
-  //   this.setState({
-  //     users : [...this.state.users, this.state.newUser],
-  //     newUser : ''
-  //   })
-  // }
-  //
-  // onDelete = (index) => {
-  //   this.setState({users : this.state.users.filter((u, i) => { return i !== index }) })
-  //   this.inputField.current.focus()
-  // }
+  addRandomContact = () => {
+    const contacts = this.state.contactsPool;
+    let randomContact = contacts[Math.floor(Math.random() * contacts.length)]
+    this.setState({contacts  : [...this.state.contacts, randomContact]})
+  }
```

Comment:
Little addition I would make is check if the contact already exists in the contacts otherwise you'd get duplicated contacts.

Reply:
Totally !